src/reader.cpp

```diff
@@ -138,6 +138,34 @@ static acquired_lock<commandline_state_t> commandline_state_snapshot() {
 
 commandline_state_t commandline_get_state() { return *commandline_state_snapshot(); }
 
+void commandline_get_part(const wchar_t *current_buffer, size_t current_cursor_pos,
```

Comment:
we should change `current_cursor_pos` to `cursor_pos` here.
Similarly, `current_buffer` should probably be `buff` like in the functions from `parse_util.cpp`.